components/km271_wifi/km271_communication_components.cpp

```diff
@@ -98,19 +110,38 @@ void BuderusParamNumber::loop()
     const uint8_t keep = 0x65;
     const uint8_t data_type_warm_water = 0x0c;
     const uint8_t data_type_heating_circuit_1 = 0x07;
+    const uint8_t data_type_heating_circuit_2 = 0x08;
 
     if (this->hasPendingWriteRequest) {
         uint32_t now = millis();
         if (now - lastWriteRequest > writeConsolidationPeriod) {
-
             if (transmissionParameter == config_ww_temperature) {
-              const uint8_t message[] = { data_type_warm_water, 0x07, keep, keep, keep, (uint8_t)this->pendingWriteValue, keep, keep};
-              sendAndConfirm(message, sizeof(message));
+                const uint8_t message[] = { data_type_warm_water, 0x07, keep, keep, keep, (uint8_t)this->pendingWriteValue, keep, keep};
+                sendAndConfirm(message, sizeof(message));
             } else if(transmissionParameter == config_heating_circuit_1_design_temperature) {
                 const uint8_t message[] = { data_type_heating_circuit_1, 0x0e, keep, keep, keep, keep, (uint8_t)this->pendingWriteValue, keep};
                 sendAndConfirm(message, sizeof(message));
-            } else if(transmissionParameter == config_heating_circuit_1_target_room_temperature_day) {
-                const uint8_t message[] = { data_type_heating_circuit_1, 0x0, keep, keep, keep, (uint8_t)(this->pendingWriteValue * 2 + 0.5), keep, keep};
+            } else if(transmissionParameter == config_heating_circuit_1_room_target_temperature_day) {
+                const uint8_t message[] = { data_type_heating_circuit_1, 0x00, keep, keep, keep, (uint8_t)(this->pendingWriteValue * 2 + 0.5), keep, keep};
+                sendAndConfirm(message, sizeof(message));
+            } else if(transmissionParameter == config_heating_circuit_1_room_target_temperature_offset) {
+                const uint8_t message[] = { data_type_heating_circuit_1, 0x31, keep, keep, keep, (uint8_t)(this->pendingWriteValue * 2), keep, keep};
```

Comment:
I'd suggest adding a +0.5 here (like above). This would improve rounding precision if somebody sends values like 0.3.
```suggestion
                const uint8_t message[] = { data_type_heating_circuit_1, 0x31, keep, keep, keep, (uint8_t)(this->pendingWriteValue * 2 + 0.5), keep, keep};
```